tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -1813,7 +1991,21 @@ tensorflow::Status ConvertIdentity(
     Converter& ctx, const tensorflow::NodeDef& node_def,
     const std::vector<TRT_TensorOrWeights>& inputs,
     std::vector<TRT_TensorOrWeights>* outputs) {
-  outputs->push_back(inputs.at(0));
+  if (inputs.at(0).is_weights()) {
```

Comment:
In the PR description you mentioned a bug is fixed with Identity node, may I know what was the bug?

Reply:
The issue was that if the identity node was the first node in the engine, the input tensor would be renamed from InputPH_0 to something like Identity. This will cause runtime error when the binding for InputPH_0 can't be found. It actually looks like you fixed this in a different way: https://github.com/tensorflow/tensorflow/blob/09c215a9c134dcc52882b153973617e125ad300b/tensorflow/contrib/tensorrt/convert/convert_nodes.cc#L727
However, your solution causes none of the tensors to have their names set - by default TRT names new tensors like "(Unamed Tensor* 1)" etc. but it looks like you check for nullptr or a len of 0.

Reply:
I should mention that I am using TRT 5.0, the default tensor names may have changed